File: server/src/main/java/com/vector/quiz/modules/quiz/service/QuizService.java
package com.vector.quiz.modules.quiz.service;

import com.vector.quiz.modules.quiz.dto.request.CreateQuizRequest;
import com.vector.quiz.modules.quiz.dto.request.UpdateQuizRequest;
import com.vector.quiz.modules.quiz.dto.response.QuizResponse;

import java.util.List;

public interface QuizService {
    List<QuizResponse> getAllQuizzes();
    QuizResponse getQuizById(Long id);
    QuizResponse createQuiz(CreateQuizRequest request);
    QuizResponse updateQuiz(Long id, UpdateQuizRequest request);
    void deleteQuiz(Long id);
    QuizResponse getQuizWithQuestionsAndAnswers(Long id);
}

File: client/src/api/services/quiz.service.ts
import { unwrapResponse } from '@/utils/api.ts';
import { getApiClient } from '@/api/client.ts';
import type { CreateQuizRequest, QuizResponse, UpdateQuizRequest } from '@/types/Quiz.ts';

const baseUrl = '/quizzes';

export function getAllQuizzes() {
  return unwrapResponse<QuizResponse[]>(getApiClient().get(baseUrl));
}

export function getQuizById(quizId: number) {
  return unwrapResponse<QuizResponse[]>(getApiClient().get(`${baseUrl}/${quizId}`));
}

export function createQuiz(params: CreateQuizRequest) {
  return unwrapResponse<QuizResponse>(getApiClient().post(baseUrl, params));
}

export function updateQuiz(params: UpdateQuizRequest) {
  return unwrapResponse<QuizResponse>(getApiClient().put(`${baseUrl}/${params.id}`, params));
}

export function deleteQuiz(quizId: number) {
  return unwrapResponse<QuizResponse>(getApiClient().delete(`${baseUrl}/${quizId}`));
}
